flash/core/data/data_module.py

```diff
@@ -786,6 +786,112 @@ def from_tensors(
+            test_targets: A sequence of targets (one per test input) to use as the test targets.
+            predict_data: A torch_geometric Data object or collection of objects to use when predicting.
+            train_transform: The dictionary of transforms to use during training which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            val_transform: The dictionary of transforms to use during validation which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            test_transform: The dictionary of transforms to use during testing which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            predict_transform: The dictionary of transforms to use during predicting which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            data_fetcher: The :class:`~flash.core.data.callback.BaseDataFetcher` to pass to the
+                :class:`~flash.core.data.data_module.DataModule`.
+            preprocess: The :class:`~flash.core.data.data.Preprocess` to pass to the
+                :class:`~flash.core.data.data_module.DataModule`. If ``None``, ``cls.preprocess_cls``
+                will be constructed and used.
+            val_split: The ``val_split`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            batch_size: The ``batch_size`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            num_workers: The ``num_workers`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            preprocess_kwargs: Additional keyword arguments to use when constructing the preprocess. Will only be used
+                if ``preprocess = None``.
+
+        Returns:
+            The constructed data module.
+
+        Examples::
+
+            data_module = DataModule.from_pygdatasequence(
+                train_files=[
+                    Any(torch_geometric.utils.from_networkx(nx.complete_bipartite_graph(3,2))),
+                    Any(torch_geometric.utils.from_networkx(nx.tetrahedral_graph()))
+                    Any(torch_geometric.utils.from_networkx(nx.complete_bipartite_graph(4,1))),
+                ],
+                train_targets=[1, 0, 1],
+                train_transform={
+                    torch_geometric.transforms.Cartesian(),
+                },
+            )
+        """
+        if not train_targets and train_data:
+            train_targets = []
+            for dat in train_data:
+                train_targets.append(dat.y)
+                dat.y = None
+
+        if not val_targets and val_data:
+            val_targets = []
+            for dat in val_data:
+                val_targets.append(dat.y)
+                dat.y = None
```

Comment:
I think it would be cleaner to just have this in the `load_data` of the `DataSource` instead. Then it wouldn't need to be repeated for train, test, and val

Reply:
I think I have somewhat repeated myself: https://github.com/PyTorchLightning/lightning-flash/pull/617/files#r677785360
How would we then modify this file?